lib/ansible/modules/windows/win_domain_computer.py

```diff
@@ -0,0 +1,122 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_domain_computer
+short_description: Management of Computer Active Directory Objects
+description:
+  - >
+    Create, read, update and delete computers in Active Directory using a
+    windows brigde computer to launch New-ADComputer, Get-ADComputer,
+    Set-ADComputer, Remove-ADComputer and Move-ADObject.
+options:
+  name:
+    description:
+      - >
+        Specifies the name of the object. This parameter sets the Name property
+        of the Active Directory object. The LDAP display name (ldapDisplayName)
+        of this property is name.
+    required: true
+  sam_account_name:
+    description:
+      - >
+        Specifies the Security Account Manager (SAM) account name of the user,
+        group, computer, or service account. The maximum length of the
+        description is 256 characters. To be compatible with older operating
+        systems, create a SAM account name that is 15 characters or less. This
+        parameter sets the SAMAccountName for an account object. The LDAP
+        display name (ldapDisplayName) for this property is sAMAccountName.
+        Note: If the SAMAccountName string provided does not end with a $, a $
+        will be appended if needed.
+    default: C(name)
+  enabled:
+    description:
+      - >
+        Specifies if an account is enabled. An enabled account requires a
+        password. This parameter sets the Enabled property for an account
+        object. This parameter also sets the ADS_UF_ACCOUNTDISABLE flag of the
+        Active Directory User Account Control (UAC) attribute.
+    default: true
+  ou:
+    description:
+      - >
+        Specifies the X.500 path of the Organizational Unit (OU) or container
+        where the new object is created. Required when I(state=present).
+  description:
+    description:
+      - >
+        Specifies a description of the object. This parameter sets the value
+        of the Description property for the object. The LDAP display name
+        (ldapDisplayName) for this property is description.
+    default: ''
+  dns_host_name:
+    description:
+      - >
+        Specifies the fully qualified domain name (FQDN) of the computer. This
+        parameter sets the DNSHostName property for a computer object. The LDAP
+        display name for this property is dNSHostName. Required when
+        I(state=present).
+  state:
+    description:
+      - >
+        Specify whether to add C(present) or remove C(absent) the specified
```

Comment:
Add or remove are not declarative, but imperative. So I would rather state: Specified whether the computer should be present or absent in Active Directory.

Beware that the current description is talking about access rules, which is weird because we are managing computers in AD.

Reply:
Copy & paste weirdness :)
Done too ☑️